src/legacy/core_plugins/vis_default_editor/public/components/controls/raw_json.tsx

```diff
@@ -34,48 +32,69 @@ function RawJsonParamEditor({
   setValue,
   setTouched,
 }: AggParamEditorProps<string>) {
+  const [isFormValid, setFormValidity] = useState(true);
+  const [editorReady, setEditorReady] = useState(false);
+
+  const editorTooltipText = i18n.translate('visDefaultEditor.controls.jsonInputTooltip', {
```

Comment:
To optimize rendering of child components, I'd suggest to use `useMemo` for constants `editorTooltipText`, `jsonEditorLabelText` and `label`, and `useCallback` for `onEditorValidate` function.